pkg/scheduler/internal/cache/cache.go

```diff
@@ -569,6 +570,47 @@ func (cache *schedulerCache) RemoveNode(node *v1.Node) error {
 	return nil
 }
 
+func (cache *schedulerCache) AddCSINode(csiNode *storagev1beta1.CSINode) error {
+	cache.mu.Lock()
+	defer cache.mu.Unlock()
+
+	n, ok := cache.nodes[csiNode.Name]
+	if !ok {
+		n = newNodeInfoListItem(schedulernodeinfo.NewNodeInfo())
+		cache.nodes[csiNode.Name] = n
+	}
+	n.info.SetCSINode(csiNode)
+	cache.moveNodeInfoToHead(csiNode.Name)
+	return nil
+}
+
+func (cache *schedulerCache) UpdateCSINode(oldCSINode, newCSINode *storagev1beta1.CSINode) error {
+	cache.mu.Lock()
+	defer cache.mu.Unlock()
+
+	n, ok := cache.nodes[newCSINode.Name]
+	if !ok {
+		n = newNodeInfoListItem(schedulernodeinfo.NewNodeInfo())
+		cache.nodes[newCSINode.Name] = n
+	}
+	n.info.SetCSINode(newCSINode)
+	cache.moveNodeInfoToHead(newCSINode.Name)
+	return nil
+}
+
+func (cache *schedulerCache) RemoveCSINode(csiNode *storagev1beta1.CSINode) error {
+	cache.mu.Lock()
+	defer cache.mu.Unlock()
+
+	n, ok := cache.nodes[csiNode.Name]
+	if !ok {
+		return fmt.Errorf("node %v is not found", csiNode.Name)
+	}
+	n.info.SetCSINode(nil)
```

Comment:
If n, the NodeInfoListItem, was created on line 579, shouldn't cache.removeNodeInfoFromList be called ?